packages/@aws-cdk/core/lib/duration.ts

```diff
@@ -143,6 +143,36 @@ export class Duration {
     }
   }
 
+  /**
+   * Turn this duration into a human-readable string
+   */
+  public toHumanString(): string {
+    if (this.amount === 0) { return fmtUnit(0, this.unit); }
+    if (Token.isUnresolved(this.amount)) { return `<token> ${this.unit.label}`; }
+
+    let millis = convert(this.amount, this.unit, TimeUnit.Milliseconds, { integral: false });
+    const parts = new Array<string>();
+
+    for (const unit of [TimeUnit.Days, TimeUnit.Hours, TimeUnit.Hours, TimeUnit.Minutes, TimeUnit.Seconds, TimeUnit.Milliseconds]) {
+      const wholeCount = Math.floor(convert(millis, TimeUnit.Milliseconds, unit, { integral: false }));
+      if (wholeCount > 0) {
+        parts.push(fmtUnit(wholeCount, unit));
+        millis -= wholeCount * unit.inMillis;
+      }
+    }
+
+    // 2 significant parts, that's totally enough for humans
+    return parts.slice(0, 2).join(' ');
```

Comment:
Could be a configurable option, but we can totally do that later.

Reply:
Was thinking the same.